### ISSUE TITLE
v2rdm_casscf-v2rdm tests failing

### ISSUE THREAD

**Issue by jgonthier** [psi4/psi4#905]
Hi everyone,

I newly compiled Psi4 Rev {master} d7d117e pulled from the latest Master, with Intel MKL 2018.1.163 and GNU compilers version 5.4.0.

I used the following cmake options to try to get most of the optional modules and get CheMPS2 interprocedural optimization to work:
cmake -H. -Bobj_rel -DMAX_AM_ERI=6 -DENABLE_ambit=ON -DENABLE_CheMPS2=ON -DENABLE_dkh=ON -DENABLE_libefp=ON -DENABLE_simint=ON -DENABLE_gdma=ON -DENABLE_PCMSolver=ON -DENABLE_snsmp2=ON -DENABLE_v2rdm_casscf=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_AR=/usr/bin/gcc-ar -DCMAKE_RANLIB=/usr/bin/gcc-ranlib

Everything configures and builds like a charm, then all tests run except those that seem to end with a segfault shortly before the 2-RDM computation starts:
        401 - v2rdm_casscf-v2rdm1 (Failed)
        402 - v2rdm_casscf-v2rdm2 (Failed)
        403 - v2rdm_casscf-v2rdm3 (Failed)
        404 - v2rdm_casscf-v2rdm4 (Failed)
        405 - v2rdm_casscf-v2rdm5 (Failed)
        406 - v2rdm_casscf-v2rdm6 (Failed)

make pytest shows all non-skipped tests passed. I also see 
363 - dft-custom-hybrid (Failed)
but this one is because the GCP software is not installed so I guess the test should have been skipped in this case.

I don't actually need to run 2-RDM computations but I thought I should let you know.

**Comment by loriab** [comment]
Yeah, I actually just separated the dft-custom-functional into two tests, w/ and w/o gcp. 

Are you sure it was either building v2rdm internally or finding a recent enough (early Dec) one? Psi keeps breaking API so v2rdm keeps needing updates. I just tried it locally and v2rdm3 passes (others fail, but that's `for_realloc_lhs`, a build error), so I don't think it's wholly broken. If your psi was picking up a old v2rdm lying around, I can believe it would thoroughly fail, as you see.

Thanks for the report!
